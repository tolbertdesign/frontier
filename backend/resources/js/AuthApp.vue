<style>
:root {
  --color-primary: #2db564;
  --color-secondary: #2e86de;
  --color-tertiary: #ed8936;
  --color-danger: #d32f2f;
  --color-modal-button: #d0021b;
  --color-facebook: #3b5998;
  --color-emailer: #00a6d9;
  --color-sms: #57af7b;
  --color-copy: #00999e;
  --color-secondary-stroke: #7f8fa4;
  --color-text: #4a4a4a;
  --color-disable-gray: #efefef;
  --color-bg-tooltip: #eef2f7;
  --color-white: #f9fafc;
  --color-bg-white: #f9f9f9;
  --color-header-light: #027ed7;
  --color-header-dark: #02098a;

  /* Examples  */
  --color-bg-primary: #0252cc;
  --color-bg-secondary: #ffc402;
  --color-bg-default: #fff;
  --color-bg-inverse: #fff;
  --color-text-primary: #fff;
  --color-text-secondary: #755f17;
  --color-text-default: #444;
  --color-text-default-soft: #999;
  --color-text-inverse: #444;
  --color-text-inverse-soft: #999;
  --font-display: "Helvetica Neue", Helvetica, Arial, sans-serif;
  --font-body: "Helvetica Neue", Helvetica, Arial, sans-serif;
  --font-weight-normal: 400;
  --font-weight-display: 400;
  --font-weight-btn: 600;
  --rounded-btn: .25rem;
}
</style>
