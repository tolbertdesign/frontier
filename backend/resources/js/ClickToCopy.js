import ClipboardJS from 'clipboard'
class ClickToCopy {
  static enable () {
    $(document).ready(() => {
      const clipboardjs = new ClipboardJS('.copy-text')

      clipboardjs.on('success', (event) => {
        $('.copy-text').tooltip({ trigger: 'manual' })
        $(event.trigger).tooltip('show')

        setTimeout(function () {
          $(event.trigger).tooltip('hide')
        }, 1000)
      })
    })
  }
}
export default ClickToCopy
