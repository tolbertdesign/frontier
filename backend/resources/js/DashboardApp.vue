<template>
  <div id="app" class="w-full h-screen transition-all duration-300">
    <StoreLoader
      :user="currentUser"
      :s3_bucket="awsBucket"
      :beta_banner_kill_switch="betaBannerKillSwith"
      :min_password_length="minPassordLength"
      :avatar_path="avatarPath"
      :pledge_types="pledgeTypes"
      :sponsor_types="sponsorTypes"
      :countries="countries"
      :states="states"
      :lang="lang"
    />
    <Component :is="layout">
      <Transition name="page" mode="out-in">
        <RouterView :key="$route.fullPath" class="page" />
      </Transition>
    </Component>
    <PortalTarget name="modal-outlet" />
    <PortalTarget name="notification-outlet" />
    <PortalTarget name="debug-outlet" />
    <!-- <Consent
      message="I use Cookies for user analysis and on-page improvements!"
      link-label="Privacy Policy"
      href="https://lichter.io/privacy/"
    /> -->
  </div>
</template>

<script>
import Consent from 'vue-cookieconsent-component/src/components/CookieConsent.vue'
import appConfig from '@/app.config'
const defaultLayout = 'Default'

export default {
  name: 'DashboardApp',
  components: { Consent },
  data () {
    return {
      width: null,
      height: null,
      showNavigation: false,
      avatarPath: null,
      awsBucket: null,
      betaBannerKillSwith: null,
      countries: null,
      currentUser: null,
      lang: null,
      minPassordLength: 7,
      pledgeTypes: null,
      sponsorTypes: null,
      states: null,
    }
  },
  computed: {
    layout () {
      return (this.$route.meta.layout || defaultLayout) + 'Layout'
    },
  },
  mounted () {
    this.width = this.$el.getBoundingClientRect().width + 'px'
    this.height = this.$el.getBoundingClientRect().height + 'px'
  },
  page: {
    title: appConfig.title,
    titleTemplate (title) {
      title = typeof title === 'function' ? title(this.$store) : title
      return title ? `${title} | ${appConfig.title}` : appConfig.title
    },
    htmlAttrs: {
      lang: 'en',
    },
    base: { target: '_blank', href: '/' },
    meta: [
      { hid: 'author', name: 'author', content: appConfig.url },
      { name: 'publisher', content: appConfig.url },
      { name: 'apple-mobile-web-app-title', content: appConfig.title },
      { name: 'theme-color', content: appConfig.themeColor },
      // Facebook
      { name: 'og:title', content: appConfig.title },
      { name: 'og:description', content: appConfig.description },
      { name: 'og:type', content: 'website' },
      { name: 'og:url', content: appConfig.url },
      { name: 'og:image', content: appConfig.img },
      { name: 'og:locale', content: appConfig.locale },
      // Twitter
      { name: 'twitter:card', content: 'summary_large_image' },
      { name: 'twitter:site', content: appConfig.twitter },
      { name: 'twitter:creator', content: appConfig.twitter },
      { name: 'twitter:title', content: appConfig.title },
      { name: 'twitter:description', content: appConfig.description },
      { name: 'twitter:image', content: appConfig.img },
    ],
  },
}
</script>
