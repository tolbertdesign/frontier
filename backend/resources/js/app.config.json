{
    "title": "Design Code",
    "tagline": "House Arrest 2020",
    "description": "An ever-evolving and opinionated dev environment for people who want to use ... to power their ...",
    "url": "https://tolbert.design",
    "img": "/social.png",
    "locale": "en_US",
    "twitter": "design_coder",
    "themeColor": "rebeccapurple"
}
