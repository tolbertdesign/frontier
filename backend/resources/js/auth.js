import 'core-js/stable'
import 'regenerator-runtime/runtime'

/**
 * First we will load all of this project's JavaScript dependencies which
 * includes Vue and other libraries. It is a great starting point when
 * building robust, powerful web applications using Vue and Laravel.
 */

import Vue from 'vue'
import store from '@/store'
import Croppa from 'vue-croppa'
import VueConfetti from 'vue-confetti'
import VueTheMask from 'vue-the-mask'
import Snowf from 'vue-snowf'
// import AuthApp from '@/AuthApp'

Vue.use(Croppa)
Vue.use(VueConfetti)
Vue.use(VueTheMask)

window._ = require('lodash')

window.Popper = require('popper.js').default

try {
  window.$ = window.jQuery = require('jquery')
  require('bootstrap')
} catch (e) {
  // eslint-disable-next-line
  console.log(e)
}

window.axios = require('axios')
window.axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest'

const token = document.head.querySelector('meta[name="csrf-token"]')

if (token) {
  window.axios.defaults.headers.common['X-CSRF-TOKEN'] = token.content
} else {
  console.error('CSRF token not found: https://laravel.com/docs/csrf#csrf-x-csrf-token')
}

/**
 * Next, we will create a fresh Vue application instance and attach it to
 * the page. Then, you may begin adding components to this application
 * or customize the JavaScript scaffolding to fit your unique needs.
 */

// Standard inputs
Vue.component('text-input', require('@/components/inputs/Text.vue').default)
Vue.component('selection-input', require('@/components/inputs/Selection.vue').default)
Vue.component('toggle-input', require('@/components/inputs/Toggle.vue').default)

// Standard Buttons
Vue.component('back', require('@/components/buttons/Back.vue').default)
Vue.component('google-login', require('@/components/buttons/GoogleLogin.vue').default)

Vue.component('carousel-component', require('@/components/CarouselComponent.vue').default)
Vue.component('login-fields', require('@/components/LoginFields.vue').default)
Vue.component('password-reset-fields', require('@/components/PasswordResetFields.vue').default)
Vue.component('progress-bar', require('@/components/ProgressBar.vue').default)

Vue.component('welcome', require('@/components/registration/Welcome.vue').default)
Vue.component('sign-up', require('@/components/registration/SignUp.vue').default)
Vue.component('welcome-back', require('@/components/registration/WelcomeBack').default)
Vue.component('registration-form', require('@/components/registration/RegistrationForm.vue').default)
Vue.component('registration-type', require('@/components/registration/RegistrationType.vue').default)
Vue.component('school-search', require('@/components/registration/SchoolSearch.vue').default)
Vue.component('participant-form', require('@/components/registration/ParticipantForm.vue').default)
Vue.component('participant-register-success', require('@/components/registration/ParticipantRegisterSuccess.vue').default)
Vue.component('registration-handler', require('@/components/registration/RegistrationHandler.vue').default)
Vue.component('welcome-teachers', require('@/components/registration/WelcomeTeachers.vue').default)
Vue.component('upload-photo-form', require('@/components/registration/UploadPhotoForm/UploadPhotoForm.vue').default)
Vue.component('Snowf', Snowf)

// eslint-disable-next-line
const app = new Vue({
  el: '#app',
  store,
})

$(document).ready(function () {
  var trigger = $('.hamburger')
  var overlay = $('.overlay')
  var isClosed = false

  trigger.click(function () {
    hamburgerCross()
  })

  function hamburgerCross () {
    if (isClosed === true) {
      overlay.hide()
      trigger.removeClass('is-open')
      trigger.addClass('is-closed')
      isClosed = false
    } else {
      overlay.show()
      trigger.removeClass('is-closed')
      trigger.addClass('is-open')
      isClosed = true
    }
  }

  $('[data-toggle="offcanvas"]').click(function () {
    $('#wrapper').toggleClass('toggled')
  })

  $('.copy-text').tooltip({ trigger: 'manual' })
  $('.copy-text').on('click', function () {
    var that = this
    that.select()
    document.execCommand('Copy')
    $(that).tooltip('show')
    setTimeout(function () {
      $(that).tooltip('hide')
    }, 1000)
  })
})

$(function () {
  $('[data-toggle="tooltip"]').tooltip()
})
