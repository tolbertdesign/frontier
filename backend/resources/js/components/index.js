// Global Components
import Vue from 'vue'

import BaseIcon from '@/components/element/BaseIcon'
import BaseInput from '@/components/element/BaseInput'
import RoundedButton from '@/components/element/RoundedButton'
import AvatarImage from '@/components/element/AvatarImage'
import RingChart from '@/components/element/RingChart'
import ProgressMeter from '@/components/element/ProgressMeter'
import SendEnvelope from '@/components/element/SendEnvelope'
import Trophy from '@/components/element/Trophy'
import VideoIframe from '@/components/element/VideoIframe'
import ReadMoreComponent from '@/components/element/ReadMoreComponent'

Vue.component('BaseIcon', BaseIcon)
Vue.component('BaseInput', BaseInput)
Vue.component('RoundedButton', RoundedButton)
Vue.component('AvatarImage', AvatarImage)
Vue.component('RingChart', RingChart)
Vue.component('ProgressMeter', ProgressMeter)
Vue.component('SendEnvelope', SendEnvelope)
Vue.component('Trophy', Trophy)
Vue.component('VideoIframe', VideoIframe)
Vue.component('ReadMoreComponent', ReadMoreComponent)
