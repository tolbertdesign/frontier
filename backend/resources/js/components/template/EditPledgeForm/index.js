import EditPledgeForm from './EditPledgeForm'

export default EditPledgeForm
