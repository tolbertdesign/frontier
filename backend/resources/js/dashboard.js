import 'core-js/stable'
import 'regenerator-runtime/runtime'

import axios from 'axios'
import axon from '@/axon'
import Vue from 'vue'
import router from '@/router'
import store from '@/store'
import storeLoader from '@/storeLoader'

import upperFirst from 'lodash/upperFirst'
import camelCase from 'lodash/camelCase'

import '@/plugins'
import '@/components'
import '@/filters'
import '@/mixins'
import '@/utilities'

// Globally Register Base and Layout Components
const requireComponent = require.context(
  './components',
  false,
  /Base[A-Z]\w+\.(vue|js)$/,
)
requireComponent.keys().forEach(fileName => {
  const componentConfig = requireComponent(fileName)
  const componentName = upperFirst(
    camelCase(fileName.replace(/^\.\/(.*)\.\w+$/, '$1')),
  )
  Vue.component(componentName, componentConfig.default || componentConfig)
})

// TODO: Handle initialization in a better way
Vue.component('store-loader', storeLoader)

//
// require('./bootstrap');
require('@/navigation')
require('slick-carousel/slick/slick')

const CarouselCustomizations = require('@/CarouselCustomizations')
const ClickToCopy = require('@/ClickToCopy')
ClickToCopy.enable()
const GoogleAnalytics = require('@/GoogleAnalytics')
const Facebook = require('@/Facebook')

const carouselCustomizations = new CarouselCustomizations()
const googleAnalytics = new GoogleAnalytics()
const facebook = new Facebook()

carouselCustomizations.enable()
googleAnalytics.enable()
facebook.enable()

window.Vue.component('readmore-component', require('@/components/ReadMoreComponent.vue').default)
window.Vue.component('vimeo-thumbnail', require('@/components/VimeoThumbnail.vue').default)
window.Vue.component('level-set', require('@/components/LevelSet.vue').default)
//

if (window.__INITIAL_STATE__) {
  store.replaceState(window.__INITIAL_STATE__)
}

// Don't warn about using the dev version of Vue in development.
Vue.config.productionTip = process.env.NODE_ENV === 'production'

// Make Axon available globally as `this.$axon` in components
Vue.prototype.$axon = window.axon = axon
Vue.prototype.$http = window.axios = axios

window.axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest'
window.axios.defaults.withCredentials = true
window.axios.defaults.baseURL = 'http://ecosystem.test'

const token = document.head.querySelector('meta[name="csrf-token"]')
if (token) {
  window.axios.defaults.headers.common['X-CSRF-TOKEN'] = token.content
} else {
  console.error('CSRF token not found: https://laravel.com/docs/csrf#csrf-x-csrf-token')
}

// eslint-disable-next-line
const vm = new Vue({
  el: '#app',
  store,
  router,
})

require('./slickCustom.js')

// refactor this out
document.cookie = 'hide_cookie_policy=hide;expires= Fri, 31 Dec 9999 23:59:59 GMT'

// // eslint-disable-next-line
// const app = window.Vue = new Vue({
//   i18n,
//   router,
//   store,
//   render: h => h(DashboardApp),
// }).$mount('#app')

// If running inside Cypress...
if (process.env.VUE_APP_TEST === 'e2e') {
  // Attach the app to the window, which can be useful
  // for manually setting state in Cypress commands
  // such as `cy.logIn()`.
  window.__app__ = vm

  // Ensure tests fail when Vue emits an error.
  Vue.config.errorHandler = window.Cypress.cy.onUncaughtException
}
