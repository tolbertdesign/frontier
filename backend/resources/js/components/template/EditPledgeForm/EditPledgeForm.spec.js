import { shallowMount, createLocalVue } from '@vue/test-utils'
import EditPledgeForm from './EditPledgeForm.vue'
import Vuex from 'vuex'
import Vuelidate from 'vuelidate'
import { STATUS_ENTERED, STATUS_CONFIRMED, STATUS_PAID } from '@/store/modules/pledge.js'
import { STATUS_PENDING, STATUS_DENIED } from '@/store/modules/online_pending_payment.js'
import VTooltip from 'v-tooltip'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuelidate)
localVue.use(VTooltip)

describe('EditPledgeForm', () => {
  const PPU = 1
  const FLAT = 3
  const program = {
    unit_range_low: 30,
    unit_range_high: 35,
  }

  let pledgeData = {}
  let store
  let storeProps

  beforeEach(() => {
    pledgeData = []
    storeProps = {
      lang: {
        pledge_edit: {},
      },
      User: {
        id: 1,
      },
    }
    store = new Vuex.Store({
      state: storeProps,
    })
  })

  it('correctly reads if can edit amount when pledge is paid', done => {
    addPledge()
    pledgeData[0].pledge_status_id = STATUS_PAID
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.canEditAmount()).toBe(false)
    done()
  })

  it('correctly reads if can edit amount when pledge is confirmed (no laps)', done => {
    addPledge()
    pledgeData[0].pledge_status_id = STATUS_CONFIRMED
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.canEditAmount()).toBe(true)
    done()
  })

  it('correctly reads if can edit amount when pledge is confirmed (has laps)', done => {
    addPledge()
    pledgeData[0].pledge_status_id = STATUS_CONFIRMED
    pledgeData[0].participant.laps = 1
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.canEditAmount()).toBe(false)
    done()
  })

  it('correctly reads if can edit amount when non sponsor\'s pledge is just entered', done => {
    store.state.User.id = 2
    addPledge()
    pledgeData[0].pledge_status_id = STATUS_ENTERED
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.canEditAmount()).toBe(true)
    done()
  })

  it('correctly reads if can edit amount when non sponsor\'s pledge is just entered w/ pending payments', done => {
    store.state.User.id = 2
    addPledge()
    pledgeData[0].pledge_status_id = STATUS_ENTERED
    pledgeData[0].online_pending_payments = [
      {
        online_pending_payment_status_id: STATUS_PENDING,
      },
    ]
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.canEditAmount()).toBe(false)
    done()
  })

  it('correctly reads if can edit amount when non sponsor\'s pledge is just entered w/ non pending payments', done => {
    store.state.User.id = 2
    addPledge()
    pledgeData[0].pledge_status_id = STATUS_ENTERED
    pledgeData[0].online_pending_payments = [
      {
        online_pending_payment_status_id: STATUS_DENIED,
      },
    ]
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.canEditAmount()).toBe(true)
    done()
  })

  it('matches snapshot with 1 pledge', () => {
    addPledge(3)
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('does not disable states when US is selected', () => {
    addPledge()
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })
    expect(wrapper.find('#state').attributes('disabled')).toBe(undefined)
  })

  it('disables state on country change', () => {
    addPledge()
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })
    wrapper.find('#country').element.value = 'CA'
    wrapper.find('#country').trigger('change')
    expect(wrapper.find('#state').attributes('disabled')).toBe('disabled')
  })

  it('disables states when US is not selected', () => {
    addPledge()
    pledgeData[0].pledge_sponsor.country = 'UK'
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })
    expect(wrapper.find('#state').attributes('disabled')).toBe('disabled')
  })

  it('sets the proper fields', () => {
    addPledge(3)
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })
    expect(wrapper.vm.pledge.first_name).toBe(pledgeData[0].first_name)
    expect(wrapper.find('#sponsorFirstName').element.value).toBe(pledgeData[0].pledge_sponsor.first_name)
    expect(wrapper.find('#sponsorLastName').element.value).toBe(pledgeData[0].pledge_sponsor.last_name)
    expect(wrapper.find('#sponsorEmail').element.value).toBe(pledgeData[0].pledge_sponsor.email)
    expect(parseFloat(wrapper.find('#pledgeAmount').element.value)).toBe(pledgeData[0].amount)
  })

  it('matches snapshot with multiple pledges', () => {
    addPledge(3, FLAT)
    addPledge(3, FLAT)
    addPledge(3, FLAT)
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },

    })
    expect(wrapper).toMatchSnapshot()
  })

  it('calculates total for ppu pledge with decimal', done => {
    addPledge(3.01, PPU)
    addPledge(3.01, PPU)
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.estimatedTotal).toBe('$180.60 - $210.70')
    done()
  })

  it('calculates total for flat pledge with decimal', done => {
    addPledge(3.21, FLAT)
    addPledge(3.21, FLAT)
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.estimatedTotal).toBe('$6.42')
    done()
  })

  it('calculates total for ppu pledge', done => {
    addPledge(3, PPU)
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.estimatedTotal).toBe('$90 - $105')
    done()
  })

  it('calculates total for ppu pledge with laps', done => {
    addPledge(3, PPU, 20)
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.estimatedTotal).toBe('$60')
    done()
  })

  it('calculates total for multiple ppu pledge', done => {
    addPledge(3, PPU)
    addPledge(3, PPU)
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.estimatedTotal).toBe('$180 - $210')
    done()
  })

  it('calculates total for flat pledge', done => {
    addPledge(3, FLAT)
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.estimatedTotal).toBe('$3')
    done()
  })

  it('calculates total for multiple flat pledge', done => {
    addPledge(3, FLAT)
    addPledge(3, FLAT)
    const wrapper = shallowMount(EditPledgeForm, {
      localVue,
      store,
      propsData: { pledgeData, program },
    })

    expect(wrapper.vm.estimatedTotal).toBe('$6')
    done()
  })

  const addPledge = (amount, type, laps) => {
    pledgeData.push(
      {
        amount: amount,
        sponsor_type_id: 2,
        pledge_type_id: type,
        online_pending_payments: [],
        pledge_sponsor: {
          first_name: 'Ronald',
          last_name: 'Johnson',
          email: 'ronald.johnson@gmail.com1',
          phone_number: 7707707777,
          city: 'St. Louis',
          state: 'MO',
          country: 'US',
        },
        participant: {
          laps: laps,
        },
      },
    )
  }
})
