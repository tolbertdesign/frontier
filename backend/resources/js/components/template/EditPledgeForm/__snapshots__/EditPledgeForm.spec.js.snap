// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPledgeForm matches snapshot with 1 pledge 1`] = `
<form novalidate="novalidate" role="form" class="w-full bg-white p-8 rounded-xl shadow-lg min-h-full overflow-y-scroll" style="max-height: 90vh;">
  <div class="flex justify-between">
    <h2 class="text-2xl font-semibold">

    </h2> <button class="text-black self-start cursor-pointer"><span class="icon"><fontawesomeicon-stub icon="fas,times"></fontawesomeicon-stub></span></button>
  </div>
  <p class="mb-4">

  </p>
  <div class="field mb-4"><label for="sponsorFirstName" class="inline-block mb-1 pl-2 text-gray-700 text-sm">

    </label>
    <div class="control"><input id="sponsorFirstName" type="text" class="input"></div>
    <!---->
  </div>
  <div class="field mb-4"><label for="first_name" class="inline-block mb-1 pl-2 text-gray-700 text-sm">

    </label>
    <div class="control"><input id="sponsorLastName" type="text" name="last_name" class="input"></div>
    <!---->
  </div>
  <div class="field mb-4"><label for="first_name" class="inline-block mb-1 pl-2 text-gray-700 text-sm"></label>
    <div><input id="sponsorEmail" type="email" name="sponsor_email" class="input"></div>
    <!---->
  </div>
  <div class="field mb-4"><label for="first_name" class="inline-block mb-1 pl-2 text-gray-700 text-sm"></label>
    <div>
      <the-mask-stub mask="(###) ###-####" tokens="[object Object]" type="tel" name="phone_number" class="input"></the-mask-stub>
    </div>
    <!---->
  </div>
  <div class="field mb-4"><label class="inline-block mb-1 pl-2 text-gray-700 text-sm"></label>
    <div class="control">
      <div class="select w-full"><select id="state" class="input form-select w-full"></select></div>
    </div>
    <!---->
  </div>
  <div class="field mb-4"><label class="inline-block mb-1 pl-2 text-gray-700 text-sm"></label>
    <div class="control">
      <div class="select w-full"><select id="country" class="input form-select w-full"></select></div>
    </div>
    <!---->
  </div>
  <div class="field mb-4"><label class="inline-block mb-1 pl-2 text-gray-700 text-sm"></label>
    <div class="control">
      <div class="select w-full"><select class="input form-select w-full"></select></div>
    </div>
    <!---->
  </div>
  <div class="mb-4 block"><label class="inline-block mb-1 pl-2 text-gray-700 text-sm"></label>
    <div class="control">
      <div class="select w-full"><select class="input form-select w-full"></select></div>
    </div>
    <!---->
  </div>
  <div class="field mb-4"><label for="pledgeAmount" class="inline-block mb-1 pl-2 text-gray-700 text-sm">

    </label>
    <div class="control has-icons-left"><input id="pledgeAmount" type="text" name="pledge_amount" class="input form-input"> <span class="icon is-left">
                $
            </span></div>
    <!---->
  </div>
  <div class="mb-8 text-center">
    <p class="text-lg text-center">

    </p>
    <p class="text-2xl font-bold">
      $90 - $105
    </p>
  </div>
  <div class="text-center mb-6"></div>
  <div class="px-8"><button type="submit" class="button is-secondary is-rounded is-fullwidth shadow-md mb-3 bg-secondary text-white font-bold border">

    </button> <button class="button is-danger is-rounded is-fullwidth shadow-md mb-3 bg-red-700 text-white font-bold border">

    </button> <button class="button is-outlined is-rounded is-fullwidth shadow-md mb-3 bg-white text-gray-600 font-bold border">

    </button></div>
</form>
`;

exports[`EditPledgeForm matches snapshot with multiple pledges 1`] = `
<form novalidate="novalidate" role="form" class="w-full bg-white p-8 rounded-xl shadow-lg min-h-full overflow-y-scroll" style="max-height: 90vh;">
  <div class="flex justify-between">
    <h2 class="text-2xl font-semibold">

    </h2> <button class="text-black self-start cursor-pointer"><span class="icon"><fontawesomeicon-stub icon="fas,times"></fontawesomeicon-stub></span></button>
  </div>
  <p class="mb-4">

  </p>
  <div class="field mb-4"><label for="sponsorFirstName" class="inline-block mb-1 pl-2 text-gray-700 text-sm">

    </label>
    <div class="control"><input id="sponsorFirstName" type="text" class="input"></div>
    <!---->
  </div>
  <div class="field mb-4"><label for="first_name" class="inline-block mb-1 pl-2 text-gray-700 text-sm">

    </label>
    <div class="control"><input id="sponsorLastName" type="text" name="last_name" class="input"></div>
    <!---->
  </div>
  <div class="field mb-4"><label for="first_name" class="inline-block mb-1 pl-2 text-gray-700 text-sm"></label>
    <div><input id="sponsorEmail" type="email" name="sponsor_email" class="input"></div>
    <!---->
  </div>
  <div class="field mb-4"><label for="first_name" class="inline-block mb-1 pl-2 text-gray-700 text-sm"></label>
    <div>
      <the-mask-stub mask="(###) ###-####" tokens="[object Object]" type="tel" name="phone_number" class="input"></the-mask-stub>
    </div>
    <!---->
  </div>
  <div class="field mb-4"><label class="inline-block mb-1 pl-2 text-gray-700 text-sm"></label>
    <div class="control">
      <div class="select w-full"><select id="state" class="input form-select w-full"></select></div>
    </div>
    <!---->
  </div>
  <div class="field mb-4"><label class="inline-block mb-1 pl-2 text-gray-700 text-sm"></label>
    <div class="control">
      <div class="select w-full"><select id="country" class="input form-select w-full"></select></div>
    </div>
    <!---->
  </div>
  <div class="field mb-4"><label class="inline-block mb-1 pl-2 text-gray-700 text-sm"></label>
    <div class="control">
      <div class="select w-full"><select class="input form-select w-full"></select></div>
    </div>
    <!---->
  </div>
  <div class="mb-4 block"><label class="inline-block mb-1 pl-2 text-gray-700 text-sm"></label>
    <div class="control">
      <div class="select w-full"><select class="input form-select w-full"></select></div>
    </div>
    <!---->
  </div>
  <div class="field mb-4"><label for="pledgeAmount" class="inline-block mb-1 pl-2 text-gray-700 text-sm">

    </label>
    <div class="control has-icons-left"><input id="pledgeAmount" type="text" name="pledge_amount" class="input form-input"> <span class="icon is-left">
                $
            </span></div>
    <!---->
  </div>
  <div class="mb-8 text-center">
    <p class="text-lg text-center">

    </p>
    <p class="text-2xl font-bold">
      $9
    </p>
  </div>
  <div class="text-center mb-6"></div>
  <div class="px-8"><button type="submit" class="button is-secondary is-rounded is-fullwidth shadow-md mb-3 bg-secondary text-white font-bold border">

    </button> <button class="button is-danger is-rounded is-fullwidth shadow-md mb-3 bg-red-700 text-white font-bold border">

    </button> <button class="button is-outlined is-rounded is-fullwidth shadow-md mb-3 bg-white text-gray-600 font-bold border">

    </button></div>
</form>
`;
